src/series/ops/image.rs

```diff
@@ -30,8 +34,20 @@ impl Series {
 
     pub fn image_resize(&self, w: u32, h: u32) -> DaftResult<Series> {
         match self.data_type() {
-            DataType::Image(..) => Ok(self
-                .downcast_logical::<ImageType>()?
+            DataType::Image(mode) => {
+                let array = self.downcast_logical::<ImageType>()?;
+                match mode {
+                    // If the image mode is specified at the type-level (and is therefore guaranteed to be consistent
+                    // across all images across all partitions), store the resized image in a fixed shape image array,
+                    // since we'll have homogeneous modes, heights, and widths after resizing.
```

Comment:
@samster25 Added an optimization here for resizing same-mode images, where the resizing output will be a fixed-shape image column. We only do this if the user has specified an image mode when e.g. casting a column, since inferring that all images have the same mode upon decoding could produce inconsistent typing/data layouts across partitions.

Reply:
Cool! Let's make sure we match the same logic in the expression to_field code then

Reply:
@samster25 Great call out, I forgot to update that!

I had to change the `FunctionEvaluator.to_field()` API to take the underlying `Expr` as an arg (similar to the `FunctionEvaluator.evaluate()` API) so we could get access to the specified height/width args to make this determination. Changes are in this commit: https://github.com/Eventual-Inc/Daft/pull/1052/commits/c8d7b0321cacdfa8b1c7866b6bfdead4f0e9469b